### DISCUSSION TITLE
[NGINX LUA] [bug]attempt to index global 'skywalking_tracer ' (a nil value)

### DISCUSSION THREAD

**Discussion by jxplus** [apache/skywalking#11572] [Q&A] [answered]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar issues.

### Apache SkyWalking Component

Nginx Lua Agent (apache/skywalking-nginx-lua)

### What happened

I was use lua-agent in nginx，nginx v.1.24.0
when use lua-module 0.10.15, is ok.
when use lua-module 0.10.24 +resy-core, then reload nginx ok, but access error: attempt to index global 'skywalking_tracer ' (a nil value)

my configs: 
```
http {
    lua_package_path "/usr/local/nginx/lib/lua/?.lua;/usr/local/nginx/skywalking-nginx-lua-0.6.0/lib/?.lua;;";
    lua_shared_dict tracing_buffer 100m;
    init_worker_by_lua_block {
        local metadata_buffer = ngx.shared.tracing_buffer
        -- Set service name
        metadata_buffer:set('serviceName', 'nginx service')
        -- Instance means the number of Nginx deployment, does not mean the worker instances
        metadata_buffer:set('serviceInstanceName', 'nginx-instance1')
        -- type 'boolean', mark the entrySpan include host/domain
        metadata_buffer:set('includeHostInEntrySpan', false)
        -- set randomseed
        require("skywalking.util").set_randomseed()
    
        require("skywalking.client"):startBackendTimer("http://192.168.100.27:12800")
    
        skywalking_tracer = require("skywalking.tracer")
    }
    
    server {
        listen  80;
        server_name msm-test.com;

        rewrite_by_lua_block {
            skywalking_tracer:start("upstreams")
        }
        body_filter_by_lua_block {
            if ngx.arg[2] then
                skywalking_tracer:finish()
            end
        }
        log_by_lua_block {
            skywalking_tracer:prepareForReport()
        } 

        location / {
            root /export/dist;
            index index.html;
            try_files $uri $uri/ /index.html;
        } 
    }
}
```

### What you expected to happen

so i want to use newer lua-module version， 

### How to reproduce

like ·what happend codes·

### Are you willing to submit a pull request to fix on your own?

- [x] Yes I am willing to submit a pull request on my own!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wu-sheng** [comment]
It looks like `skywalking_tracer` variable is not accessible. We don't know the reason, maybe you could refer to Nginx document?

**Comment by jxplus** [comment, marked as answer]
I know thereason，thanks。
